compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -11885,18 +11898,20 @@ OMR::Z::TreeEvaluator::arraycmplenEvaluator(TR::Node * node, TR::CodeGenerator *
    TR::RegisterPair * firstPair = cg->allocateConsecutiveRegisterPair(firstLen, firstBaseReg);
    TR::RegisterPair * secondPair = cg->allocateConsecutiveRegisterPair(secondLen, secondBaseReg);
    TR::Register * resultReg = cg->allocateRegister();
+   TR::LabelSymbol * shortCircuitLabel = generateLabelSymbol(cg);
    TR::Instruction * cursor;
 
    TR::RegisterDependencyConditions * dependencies = cg->createDepsForRRMemoryInstructions(node, firstPair, secondPair);
 
+   generateRRInstruction(cg, TR::InstOpCode::getLoadRegOpCode(), node, resultReg, orgLen);
```

Comment:
Do you have a selection log that executes this path? If not can you explain how moving loading the origLen before CLCL different compared to after executing CLCL?

Reply:
In case that addresses are equal and we need to short circuit, the result should be equal to the length since this is `arraycmplen()` evaluator and we return the address of the first mismatch!